components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/vault/VaultTool.java

```diff
@@ -195,4 +218,22 @@ private static String doEncryption(Cipher cipher, String plainTextPwd) {
         System.out.println("\nEncryption is done Successfully\n");
         return encodedValue;
     }
+
+    private static void addSystemProperties() {
+        Properties properties = new Properties();
+        String filePath = System.getProperty(CONF_DIRECTORY_PATH) + File.separator + CARBON_PROPERTIES;
+        File file = new File(filePath);
+
+        if (file.exists()) {
+            try ( InputStream inputStream = new FileInputStream(file)) {
+                properties.load(inputStream);
+            } catch (IOException e) {
+                e.printStackTrace();
+                System.exit(1);
+            }
+        }
+
+        String secureVaultRegex = properties.getProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY);
+        System.setProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY, secureVaultRegex);
```

Comment:
Null check must be done here. 

Related PR : https://github.com/wso2/micro-integrator/pull/841

Reply:
Yes it needs, Otherwise it will try to put a null value to the HashMap when secureVaultRegex value is null.